Iteration 7

aliciatay-zls opened the pull request:
Complete to-do list by mentor:
- [x] Implement same interface for McDonald's and Fairprice parsers
- [x] Add debug mode for parsers
- [x] Handle empty parameters in rest_api.js
- [ ] Add to README.md
- [x] Use CSS framework, work on front-end

Extra:
- [ ] Add new sample data: a restaurant
- [ ] Add custom error class, improve error handling
- [x] Refactor rest_api.js

aliciatay-zls committed:
Implement same interface for McDonald's parser

aliciatay-zls committed:
Implement same interface for Fairprice parser

aliciatay-zls committed:
Fix geoGetCurrentLocation() in locations.js

miccheng commented on html/locations.js:
```diff
@@ -8,8 +8,8 @@ async function geoGetCurrentLocation() {
-        currentLatitude = pos.coords.longitude;
-        currentLongitude = pos.coords.latitude;
+        currentLatitude = pos.coords.latitude;
+        currentLongitude = pos.coords.longitude;
```
LOL... oops...

aliciatay-zls committed:
Handle empty/invalid parameters before sending to server

aliciatay-zls committed:
Change and fix indentation

miccheng commented:
You want to consider using something like ESLint for auto-formatting your code? 

- ESLint: https://eslint.org/docs/user-guide/getting-started
- VSCode Extension: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

aliciatay-zls committed:
Use ESLint, fix problems identified (undef vars)

aliciatay-zls commented:
> You want to consider using something like ESLint for auto-formatting your code?
> 
> * ESLint: https://eslint.org/docs/user-guide/getting-started
> * VSCode Extension: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint


Thanks Mr Cheng 😀 Done in 41ab441

aliciatay-zls committed:
Add debug mode for parsers

- make error messages more detailed
- remove ability to run seed files (no longer search by cat/brand)

aliciatay-zls committed:
Fix WHERE clause, add check for lat long in rest_api.js

- minor code refactoring

aliciatay-zls committed:
Change Haversine function to in-line MySQL query

aliciatay-zls committed:
Minor code refactoring for db_manager.js

aliciatay-zls committed:
Refactor rest_api.js

aliciatay-zls committed:
Use CSS framework

aliciatay-zls committed:
Prepare for deployment with mentor
